node/node.go

```diff
@@ -1167,6 +1168,9 @@ func (n *Node) Close() error {
 
 	var closeError error = nil
 
+	n.mutClosableComponents.Lock()
```

Comment:
RLock/RUnlock and make mutClosableComponents a RWMutex

Reply:
+1